modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsRepositoryImpl.java

```diff
@@ -46,26 +55,38 @@
      *
      * @param storeData If {@code true} - node stores data locally, {@code false} - otherwise.
      * @param persistence If {@code true} - node have persistence store, {@code false} - otherwise.
+     * @param subscriptionProcessor Subscription processor.
      * @param statisticsManager Ignite statistics manager.
      * @param log Ignite logger to use.
      */
     public IgniteStatisticsRepositoryImpl(
             boolean storeData,
             boolean persistence,
+            IgniteCacheDatabaseSharedManager database,
+            GridInternalSubscriptionProcessor subscriptionProcessor,
             IgniteStatisticsManagerImpl statisticsManager,
-            IgniteLogger log) {
+            Function<Class<?>, IgniteLogger> logSupplier
+    ) {
         if (storeData) {
             // Persistence store
-            partsStats = (persistence) ? null : new ConcurrentHashMap<>();
+            if (persistence) {
+                store = new IgniteStatisticsStoreImpl(subscriptionProcessor, database,this, logSupplier);
+                partsStats = null;
+            }
+            else {
+                store = null;
+                partsStats = new ConcurrentHashMap<>();
+            }
             localStats = new ConcurrentHashMap<>();
         }
         else {
             // Cache only global statistics, no store
+            store = null;
             partsStats = null;
```

Comment:
let's get rid of all these local collections in favor of a in-memory store implementation

Reply:
Store won't keep local and global lvl of statistics. We can implement it there, but in my opinion store should be just easy wrapper around metastorage or in memory data structure, while repository should care some logic with updating and call manager to aggregate statistics on startup.

Reply:
Now I see. Yes, agree with you